Sources/StreamChatUI/ChatChannelList/SwipeableView.swift

```diff
@@ -0,0 +1,203 @@
+
+        actionStackViewWidthConstraint = actionItemsStackView.widthAnchor.pin(equalToConstant: startValue)
+        actionStackViewWidthConstraint?.priority = .required
+        actionStackViewWidthConstraint?.isActive = true
+
+        actionItemsStackView.axis = .horizontal
+        actionItemsStackView.alignment = .fill
+        actionItemsStackView.distribution = .fillProportionally
+    }
+
+    @objc open func didPan(_ gesture: UIPanGestureRecognizer) {
+        // If we don't have indexPath or any actionViews, we don't want to proceed with the swiping.
+        guard let indexPath = indexPath, let actionButtons = delegate?.swipeableViewActionViews(for: indexPath) else { return }
+
+        if gesture.state == .began {
+            // If actionButtonStackView is not open, create the action buttons to show.
+            if !isOpen {
+                // Prepare the delegate that actionViews will be shown on some cell to
+                // reset other cells states or do some other cleanup.
+                delegate?.swipeableViewWillShowActionViews(for: indexPath)
+
+                // Remove all subviews and add them to prevent having duplicities for views.
+                actionItemsStackView.removeAllArrangedSubviews()
+                actionButtons.forEach { actionItemsStackView.addArrangedSubview($0) }
+            }
+            startValue = actionStackViewWidthConstraint?.constant ?? 0
+        }
+
+        var swipeVelocity = gesture.velocity(in: self)
+        var swipePosition = gesture.translation(in: self)
+
+        // If the language is Right to left, we need to switch the swipe direction, so we just negate it.
+        if !currentLanguageIsRightToLeftDirection {
+            swipePosition.x.negate()
+            swipeVelocity.x.negate()
+        }
+
+        // We continue with the swipe and assing how far we are to `swipedValue`.
+        let swipedValue: CGFloat
+        if swipePosition.x < 0 {
+            swipedValue = startValue + swipePosition.x
+        } else {
+            swipedValue = max(0, startValue + swipePosition.x)
+        }
+
+        // If gesture failed or canceled, we close the action buttons
+        if gesture.state == .cancelled || gesture.state == .failed {
+            actionStackViewWidthConstraint?.constant = 0
+            return
+        }
+
+        // If everything went well and we swiped more than the width of the `actionStackView`,
+        // we reveal the action items animated, else we just close
+        if gesture.state == .ended {
+            let idealWidth = actionItemsStackView.idealWidth
+
+            if swipeVelocity.x < -30 {
+                actionStackViewWidthConstraint?.constant = 0
+            } else if swipedValue > idealWidth / 2.0 || swipeVelocity.x > 30 {
+                actionStackViewWidthConstraint?.constant = idealWidth
+            } else {
+                actionStackViewWidthConstraint?.constant = 0
+            }
```

Comment:
- we should maybe put the magic `30` into a variable?
- does this work correctly for RTL languages?